cf/field.py

```diff
@@ -4749,6 +4748,222 @@ def radius(self, default=None):
+        *one_sided_at_boundary* parameters). If missing values are
+        present then missing values will be returned at all points
+        where a centred finite difference could not be calculated.
+
+        .. versionadded:: 3.12.0
+
+        .. seealso:: `derivative`, `grad_xy`, `iscyclic`, `cf.div_xy`
+
+        :Parameters:
+
+            x_wrap: `bool`, optional
+                Whether the X axis is cyclic or not. By default
+                *x_wrap* is set to the result of this call to the
+                field construct's `iscyclic` method:
+                ``f.iscyclic('X')``. If the X axis is cyclic then
+                centred differences at one boundary will always use
+                values from the other boundary, regardless of the
+                setting of *one_sided_at_boundary*.
+
+                The cyclicity of the Y axis is always set to the
+                result of ``f.iscyclic('Y')``.
+
+            one_sided_at_boundary: `bool`, optional
+                If True then one-sided finite differences are
+                calculated at the non-cyclic boundaries. By default
+                missing values are set at non-cyclic boundaries.
+
+            {{radius: optional}}
+
+            {{inplace: `bool`, optional}}
+
+        :Returns:
+
+            `Field` or `None`
+                The horizontal Laplacian of the scalar field, or
+                `None` if the operation was in-place.
+
+        >>> f = cf.example_field(0)
+        >>> print(f)
+        Field: specific_humidity (ncvar%q)
+        ----------------------------------
+        Data            : specific_humidity(latitude(5), longitude(8)) 1
+        Cell methods    : area: mean
+        Dimension coords: latitude(5) = [-75.0, ..., 75.0] degrees_north
+                        : longitude(8) = [22.5, ..., 337.5] degrees_east
+                        : time(1) = [2019-01-01 00:00:00]
+        >>> f[...] = 0.1
+        >>> print(f.array)
+        [[0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1]
+         [0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1]
+         [0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1]
+         [0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1]
+         [0.1 0.1 0.1 0.1 0.1 0.1 0.1 0.1]]
+        >> >lp = f.laplacian_xy(radius='earth')
+        >>> lp
+        <CF Field: long_name=X-Y Laplacian of specific_humidity(latitude(5), longitude(8)) m-2.rad-2>
+        >>> print(lp.array)
+        [[-- -- -- -- -- -- -- --]
+         [-- -- -- -- -- -- -- --]
+         [0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0]
+         [-- -- -- -- -- -- -- --]
+         [-- -- -- -- -- -- -- --]]
+        >>> lp = f.laplacian_xy(radius='earth', one_sided_at_boundary=True)
+        >>> print(lp.array)
+        [[0. 0. 0. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 0. 0. 0.]]
+
+        """
+        from numpy import pi
+
+        f = _inplace_enabled_define_and_cleanup(self)
```

Comment:
In-place Laplacian calculation doesn't seem to work properly, though I can't immediately see why since the standard in-place logic and decorator are all there - any ideas what is going on? I see:

```pycon
>>> import cf
>>> a = cf.example_field(0)
>>> print("A IS", a, a.array) 
>>> b = a.laplacian_xy(radius="earth")
>>> print("B IS", b, b.array) 
>>> c = a.laplacian_xy(radius="earth", inplace=True)
>>> print("IN-PLACE A IS", a, a.array, , "C IS" c)
>>> cf.equals(a, b, verbose=3)
A IS Field: specific_humidity (ncvar%q)
----------------------------------
Data            : specific_humidity(latitude(5), longitude(8)) 1
Cell methods    : area: mean
Dimension coords: latitude(5) = [-75.0, ..., 75.0] degrees_north
                : longitude(8) = [22.5, ..., 337.5] degrees_east
                : time(1) = [2019-01-01 00:00:00] [[0.007 0.034 0.003 0.014 0.018 0.037 0.024 0.029]
 [0.023 0.036 0.045 0.062 0.046 0.073 0.006 0.066]
 [0.11  0.131 0.124 0.146 0.087 0.103 0.057 0.011]
 [0.029 0.059 0.039 0.07  0.058 0.072 0.009 0.017]
 [0.006 0.036 0.019 0.035 0.018 0.037 0.034 0.013]]
B IS Field: long_name=Horizontal Laplacian of specific_humidity (ncvar%q)
--------------------------------------------------------------------
Data            : long_name=Horizontal Laplacian of specific_humidity(latitude(5), longitude(8)) m-2.rad-2
Cell methods    : area: mean
Dimension coords: latitude(5) = [-75.0, ..., 75.0] degrees_north
                : longitude(8) = [22.5, ..., 337.5] degrees_east
                : time(1) = [2019-01-01 00:00:00] [[-- -- -- -- -- -- -- --]
 [-- -- -- -- -- -- -- --]
 [-2.143064530824368e-15 -2.6749442771355633e-15 -2.4238409002370553e-15
  -2.6377524366954036e-15 -1.0992306748059246e-15 -1.6075152632499628e-15
  3.5426406417251047e-16 2.2860907541031372e-15]
 [-- -- -- -- -- -- -- --]
 [-- -- -- -- -- -- -- --]]
IN-PLACE A IS Field: specific_humidity (ncvar%q)
----------------------------------
Data            : specific_humidity(latitude(5), longitude(8)) 1
Cell methods    : area: mean
Dimension coords: latitude(5) = [-1.3089969389957472, ..., 1.3089969389957472] radians
                : longitude(8) = [0.39269908169872414, ..., 5.8904862254808625] radians
                : time(1) = [2019-01-01 00:00:00] [[0.007 0.034 0.003 0.014 0.018 0.037 0.024 0.029]
 [0.023 0.036 0.045 0.062 0.046 0.073 0.006 0.066]
 [0.11  0.131 0.124 0.146 0.087 0.103 0.057 0.011]
 [0.029 0.059 0.039 0.07  0.058 0.072 0.009 0.017]
 [0.006 0.036 0.019 0.035 0.018 0.037 0.034 0.013]] C IS None
Field: Different Units: <Units: 1> != <Units: m-2.rad-2>
False
```

Reply:
Ah - got it. The culprit is that `f` is reassigned:
```python 
 f = _inplace_enabled_define_and_cleanup(self)
 <snip>
 f = d2f_dx2 + d2f_dy2
 return f
```
I'll fix that and add the test that caught it.

Reply:
Hmm. Weird things started happening with NaNs when it was properly in-place. I think the answers is to remove `inplace` from `laplacian_xy` for now, and see if anyone ever notices (there would be no real performance benefit from it, in this case, anyway)

Reply:
(Also, none of the other new methods/function have inplace, so it sort of consistent)

Reply:
> I think the answer is to remove inplace from laplacian_xy for now, and see if anyone ever notices (there would be no real performance benefit from it, in this case, anyway)

> (Also, none of the other new methods/function have inplace, so it sort of consistent)

Sounds good to me! If anyone asks it is for consistency...